contractcourt/breach_resolver.go

```diff
@@ -0,0 +1,121 @@
+package contractcourt
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+)
+
+// breachResolver is a resolver that will handle breached closes. In the
+// future, this will likely take over the duties the current breacharbiter has.
+type breachResolver struct {
+	// resolved reflects if the contract has been fully resolved or not.
+	resolved bool
+
+	// subscribed denotes whether or not the breach resolver has subscribed
+	// to the breacharbiter for breach resolution.
+	subscribed bool
+
+	// replyChan is closed when the breach arbiter has completed serving
+	// justice.
+	replyChan chan struct{}
+
+	contractResolverKit
+}
+
+// newBreachResolver instantiates a new breach resolver.
+func newBreachResolver(resCfg ResolverConfig) *breachResolver {
+	r := &breachResolver{
+		contractResolverKit: *newContractResolverKit(resCfg),
+		replyChan:           make(chan struct{}),
```

Comment:
Could safely be buffered to a size of 1.

Reply:
I opted to close it, do you think it should be passed a struct{}{} instead?